pinot-spi/src/main/java/org/apache/pinot/spi/utils/CommonConstants.java

```diff
@@ -362,6 +365,11 @@ public static class QueryOptionKey {
         public static final String MAX_ROWS_IN_JOIN = "maxRowsInJoin";
         public static final String JOIN_OVERFLOW_MODE = "joinOverflowMode";
 
+        // Indicates the maximum length of the serialized response per server for a query. The response size that a
+        // server can return is allocated by dividing the total budget CONFIG_OF_MAX_QUERY_RESPONSE_SIZE_BYTES across
+        // all servers processing the query.
+        public static final String MAX_SERIALIZED_RESPONSE_LENGTH_PER_SERVER = "maxSerializedResponseLengthPerServer";
```

Comment:
The limit that is exposed to the user  as a config is at the query level and we internally compute the budget per server. Correct ?

In that case, why should this per server limit be put in the CommonConstants file ?

Reply:
That is correct. I've answered the reasoning in another comment above.

There's a broker instance config that defines what the maximum response size for single query can be. If this is set, this total value is divided by the number of routed servers to determine the per_server_budget. This per_server_budge is set as a query config very similar to how we handle timeout today (the alternative is to make thrift changes to pass this info).